IdentityCore/src/cache/key/MSIDLegacyTokenCacheKey.m

```diff
@@ -91,9 +91,15 @@ + (MSIDLegacyTokenCacheKey *)queryWithAuthority:(NSURL *)authority
                                        resource:(NSString *)resource
                                    legacyUserId:(NSString *)legacyUserId
 {
-    NSString *service = [self.class serviceWithAuthority:authority
-                                                resource:resource
-                                                clientId:clientId];
+    NSString *service = nil;
+    
+    if (authority
+        && clientId)
+    {
+        service = [self.class serviceWithAuthority:authority
+                                          resource:resource
+                                          clientId:clientId];
+    }
```

Comment:
I am curious, is this part just to make code more robust, or there is indeed a scenario where the condition could be false? :)

Reply:
@jasoncoolmax it is used here: https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/1168/files#diff-09e5982c5acc2d706ac52ac7e583ae96R160

Reply:
Basically, because all of them are nullable, we don't want to create service unless both authority and clientId are provided.